Add concrete usage examples to docs

dominiklohmann opened the pull request:
Note: This is still heavily work-in-progress, not worth reviewing yet.

Each subcommand currently has a `help` and a `documentation` subcommand. This change adds concrete usage example to _every_ subcommands documentation subcommand.

tobim commented:
It might be useful to split the examples into a separate string so we are flexible with regards to rendering that stuff.

dominiklohmann commented:
> It might be useful to split the examples into a separate string so we are flexible with regards to rendering that stuff.

Note that this would have implications for our docs pipeline, too. Currently, the documentation is read from Markdown files in the docs/cli subdirectory. Expanding this can obviously be done, but we need to be aware that this would increase the amount of necessary work that goes into this.

tobim commented:
> Note that this would have implications for our docs pipeline, too. Currently, the documentation is read from Markdown files in the docs/cli subdirectory. Expanding this can obviously be done, but we need to be aware that this would increase the amount of necessary work that goes into this.

Yeah, I forgot about that. In that case let's not incur that overhead on ourselves until we have need for it. I also think that it might be more appropriate to keep the documentation in one `.md` file for each command and use suitable tools to work with them.

dominiklohmann commented:
[ch14389]

lava commented:
As a general comment, the current documentation seems to rely quite heavily on the presence of a markdown post-processor. This can make the terminal output very cluttered, where the raw markdown is being printed.

I think the readability could be improved a lot for this use case by staying a bit closer to how e.g. manpage look like, while still having valid markdown. As a quick mock-up:

![format_comparison](https://user-images.githubusercontent.com/51410/80576414-6d36d100-8a05-11ea-8fc8-1d17a50f193d.png)

This doesn't need to be changed as part of this PR, but I think something good to keep in mind in the longer term.

mavam commented:
@lava does the `[x]` style of link creation work with most Markdown processors? It's a different style of linking that seems more appropriate for plain ASCII, LaTeX, etc. I think the goal should be to create the right markup, so that we can embed it at http://docs.tenzir.com. If someone finds the markup too much in the way, it's easy to just pipe the output to `mdv`. I'd rather have GFM-style code highlighting than optimizing for readability of a subset of Markdown.

dominiklohmann commented:
For the code snippets, if we indent with 4 spaces instead of using triple backticks we get no highlighting on the documentation page. 

For the links we can do this:

```
Some [text with a link][some-slug].

[some-slug]: https://some.url
```

lava commented:
@mavam I'm not sure that piping to mdv enhances the output:

![mvd](https://user-images.githubusercontent.com/51410/80615992-ae000b80-8a40-11ea-96bb-f9d024888d42.png)

mavam commented:
Is this a modified version of the docs or the one with footnote-style links? It looks like the latter, but my point was that regular links, as well as using triple-backticks for code, doesn't sacrifice readability when piping to `mdv` and friends.

lava commented:
@mavam This is the original version of the docs, i.e. the left-hand side in the first picture. The right-hand side version actually looks reasonably well when piped through mdv, although still worse than the raw text because the code block is lost as well.

Ironically, mdv seems to convert links to use out-of-line definitions by default, probably that's why it looked similar to the modified version to you. (i'm using the latest `mdv` installed via `pip3 install mdv`)

lava commented on doc/cli/vast-pivot.md:
```diff
@@ -1,13 +1,24 @@
+The `vast pivot` command retrieves data of a related type. It inspects each
+event in a query result to find an event of the requested type. If the related
+type exists in the schema, VAST will dynamically create a new query to fetch the
 contextual data according to the type relationship.
```
The first paragraph is still pretty hard to parse without roughly knowing how pivot works already. In particular, the first two sentences give the impression that pivot is creating a subset of the query result containing only events of the requested type. (and even before that one has to make the mental connection between "a related type" and "the requested type")

The third sentence also immediately raises the question of what happens when the related type does not exist in the schema, which is never answered.

lava commented on doc/cli/vast-stop.md:
```diff
@@ -1 +1,9 @@
+The `vast stop` command gracefully brings down a VAST node, and is the analog of
+the `vast start` command.
+
+WHile it is technically possible to shut down a VAST node gracefully by sending
```
Random nit: `s/WHile/While/`

dominiklohmann commented:
I know it's been almost half a year since we dropped this for other stuff, but it's finally ready for review.

Note that I've removed some of the options listing from the explore command, since that is mirrored in the helptext already.

mavam commented on doc/cli/vast-export.md:
````diff
@@ -5,4 +5,27 @@ expression. The export format must be explicitly specified:
+
+```
+vast export --max-events=100 --continuous json '#timestamp < 1 hour ago'
+```
````
In addition to an example invocation, it would also be nice to have *output* examples here.

mavam commented on doc/cli/vast-start.md:
```diff
@@ -1,3 +1,23 @@
+the `vast.conf` file, such that they are picked up by all client commands as
+well.
+
+The `node` (sometimes called `server` or `server process`) is the core piece of
```
Actually, the node is not the same as the server. See [our docs](https://docs.tenzir.com/vast/architecture/system-structure) what I mean. We use "server" when talking about the entire `vast` process that runs via `vast start`. The "client" is the other process to access the server. The "node" is the main container actor inside the server that contains everything else.

mavam replied:
I suggest using the wording as in the referenced docs.

mavam commented on doc/cli/vast-stop.md:
```diff
@@ -1 +1,9 @@
+
+While it is technically possible to shut down a VAST node gracefully by sending
+`SIGINT(2)` to the `vast start` process, it is recommended to use `vast stop` to
+shut down the node, as it works over the wire as well and guarantees a proper
```
```suggestion
shut down the server, as it works over the wire as well and guarantees a proper
```

mavam commented on doc/cli/vast-stop.md:
```diff
@@ -1 +1,9 @@
+While it is technically possible to shut down a VAST node gracefully by sending
+`SIGINT(2)` to the `vast start` process, it is recommended to use `vast stop` to
+shut down the node, as it works over the wire as well and guarantees a proper
+shutdown. The command blocks execution until the node has quit, and returns a
```
```suggestion
shutdown. The command blocks execution until the server has quit, and returns a
```

There are probably a few more node vs. server terminology fixups needed.

dominiklohmann commented:
> * When did you choose to document the respective short options? Sometimes you write "`--node` / `-N`" and sometimes you only use the long version.

I've moved to using long options everywhere.
